hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/EventCondition.java

```diff
@@ -84,9 +99,12 @@ public void setExpression(String expression) {
     }
 
     public boolean match(Event value) {
-        if (null == expression || expression.isEmpty() || null == value) {
+        if (null == value) {
             return false;
         }
+        if (null == expression || expression.isEmpty()) {
+            return true;
+        }
         String[] expressions = expression.split(",");
```

Comment:
It's not clear from any jdoc that the "expression" field can actually contain a comma-separated list of individual expression items.  Also, a comma separator is a bit dangerous given all of the string manipulations/operations that are allowed here.  I could easily see a problem occurring when a comma shows up in a string used for a value or pattern.